### PR TITLE
resourcemange: fix delete more workers than expected

### LINKED ISSUE #40893: delete more worker than expected when to downclock pool
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

delete more worker than expected when to downclock pool.

First, you need to overclock the pool many times. then to downclock, it will happen this bug.

### 2. What did you expect to see? (Required)

delete one worker

### 3. What did you see instead (Required)

delete multi worker

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #40893

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: util/gpool/spmc/spmcpool.go [hunk 1]
```diff
@@ -137,14 +137,18 @@ func (p *Pool[T, U, C, CT, TF]) Tune(size int) {
...
 	p.capacity.Store(int32(size))
 	if size > capacity {
 		for i := 0; i < size-capacity; i++ {
 			if tid, boostTask := p.taskManager.Overclock(); boostTask != nil {
 				p.addWaitingTask()
-				p.taskManager.AddSubTask(tid, boostTask.Clone())
-				p.taskCh <- boostTask
+				newTask := boostTask.Clone()
+				p.taskManager.AddSubTask(tid, newTask)
+				p.taskCh <- newTask
```
Comment (hawkingrei) [inline util/gpool/spmc/spmcpool.go:151]: This is an bug which it doesn't add copied object into the task queue

File: resourcemanager/pooltask/task_manager_scheduler.go [hunk 2]
```diff
@@ -16,10 +16,10 @@ package pooltask
 
 // Overclock is to increase the concurrency of pool.
 func (t *TaskManager[T, U, C, CT, TF]) Overclock() (tid uint64, task *TaskBox[T, U, C, CT, TF]) {
-	if t.concurrency > t.running.Load() {
-		return t.getBoostTask()
+	if t.running.Load()+1 >= int32(t.SubTaskCnt(tid)) {
+		return
 	}
-	return 0, nil
+	return t.getBoostTask()
```
Comment (hawkingrei) [inline resourcemanager/pooltask/task_manager_scheduler.go:22]: The old logic is not very clear. If tasks in the queue are ready to run, it is unnecessary to create a task.

**Review by YangKeao — APPROVED** [review]
LGTM

**Review by Defined2014 — APPROVED** [review]

**Pull request merged**